### ISSUE TITLE
Accelerating Stochastic Gradient Descent via Online Learning to Sample

### ISSUE THREAD

**Issue by robertsdionne** [BVLC/caffe#2702]
Implement a new solver that learns how to sample the most useful training examples during training.

Accelerating Stochastic Gradient Descent via Online Learning to Sample
Authors: Guillaume Bouchard, Théo Trouillon, Julien Perez, Adrien Gaidon
[Abstract](http://arxiv.org/abs/1506.09016)
[Full text](http://arxiv.org/pdf/1506.09016v1.pdf)

"SGD converged in 45 epochs (outside the graph), whereas AW-SGD converged to the same performance in 10 times less epochs for a 982% improvement in training time." (see figure 1)

**Comment by bhack** [comment]
:+1:

**Comment by Razcle** [comment]
Any update on this?

**Comment by shelhamer** [comment]
Closing since AW-SGD has not caught on.
